File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F17.P1.cpp
#include <cstdio>
#include <fstream>
#include <iterator>
#include <stdexcept>
#include <string>

// BEGIN PROBE F17.P1
std::size_t read_all() {
    std::ifstream f("data.txt");
    if (!f) throw std::runtime_error("open failed");
    std::string text{std::istreambuf_iterator<char>(f), {}};
    return text.size();
}
// END PROBE F17.P1

int main() {
    std::printf("%zu\n", read_all());
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F17.P1.rs
use std::fs::File;
use std::io;

// BEGIN PROBE F17.P1
fn read_all() -> io::Result<usize> {
    let file = File::open("data.txt")?;
    let text = io::read_to_string(&file)?;
    Ok(text.len())
}
// END PROBE F17.P1

fn main() {
    println!("{}", read_all().unwrap());
}
